contract-examples/test/fee_manager.ts

```diff
@@ -9,257 +9,89 @@ import {
-    })
-    await tx.wait()
-
-    tx = await owner.sendTransaction({
-      to: nonEnabled.address,
-      value: ethers.utils.parseEther("1")
-    })
-    await tx.wait()
-  });
-
-  it("should add contract deployer as owner", async function () {
-    const contractOwnerAddr: string = await contract.owner()
-    expect(owner.address).to.be.equal(contractOwnerAddr)
-  });
-
-  it("contract should not be able to change fee without enabled", async function () {
-    let contractRole = await managerPrecompile.readAllowList(contract.address);
-    expect(contractRole).to.be.equal(ROLES.NONE)
-    try {
-      let tx = await contract.enableWAGMIFees()
-      await tx.wait()
-    }
-    catch (err) {
-      return
-    }
-    expect.fail("should have errored")
-  })
-
-  it("contract should be added to manager list", async function () {
-    let adminRole = await ownerPrecompile.readAllowList(adminAddress);
-    expect(adminRole).to.be.equal(ROLES.ADMIN)
-    let contractRole = await ownerPrecompile.readAllowList(contract.address);
-    expect(contractRole).to.be.equal(ROLES.NONE)
-
-    let enableTx = await ownerPrecompile.setEnabled(contract.address);
-    await enableTx.wait()
-    contractRole = await ownerPrecompile.readAllowList(contract.address);
-    expect(contractRole).to.be.equal(ROLES.ENABLED)
-  });
-
-  it("admin should be able to change fees through contract", async function () {
-    const testFees = {
-      gasLimit: 12_345_678, // gasLimit
-      targetBlockRate: 2, // targetBlockRate
-      minBaseFee: 1_234_567, // minBaseFee
-      targetGas: 100_000_000, // targetGas
-      baseFeeChangeDenominator: 48, // baseFeeChangeDenominator
-      minBlockGasCost: 0, // minBlockGasCost
-      maxBlockGasCost: 10_000_000, // maxBlockGasCost
-      blockGasCostStep: 100_000 // blockGasCostStep
-    }
-    var res = await contract.getCurrentFeeConfig()
-    expect(res.gasLimit).to.be.not.equal(testFees.gasLimit)
-    expect(res.minBaseFee).to.be.not.equal(testFees.minBaseFee)
-
-    let enableTx = await contract.enableCustomFees(testFees)
-    let txRes = await enableTx.wait()
-
-    var res = await contract.getCurrentFeeConfig()
-    expect(res.gasLimit).to.be.equal(testFees.gasLimit)
-    expect(res.minBaseFee).to.be.equal(testFees.minBaseFee)
-
-    var res = await contract.getFeeConfigLastChangedAt()
-    expect(res).to.be.equal(txRes.blockNumber)
+    return ethers.getContractFactory("ExampleFeeManagerTest", { signer })
+      .then(factory => factory.deploy())
+      .then(contract => {
+        this.testContract = contract
+        return contract.deployed().then(() => contract)
+      })
+      .then(contract => contract.setUp())
+      .then(tx => Promise.all([feeManagerPromise, tx.wait()]))
+      .then(([feeManager]) => feeManager.setAdmin(this.testContract.address))
+      .then(tx => tx.wait())
   })
 
-  it("admin should be able to enable wagmi fees through contract", async function () {
```

Comment:
Enabling wagmi fees or cchain fees calls the same precompile functionality with different values.
Same functionality is covered here: `admin should be able to enable change fees` (new test)